source/services/encryption/test/EncryptionService.test.tsx

```diff
@@ -0,0 +1,194 @@
+import { NativeModules } from "react-native";
+import deepCopyViaJson from "../../../helpers/Objects";
+import { AnsweredForm } from "../../../types/Case";
+import {
+  EncryptionErrorStatus,
+  EncryptionType,
+} from "../../../types/Encryption";
+import StorageService from "../../StorageService";
+import {
+  EncryptionException,
+  getStoredSymmetricKey,
+  UserInterface,
+} from "../EncryptionHelper";
+import {
+  decryptFormAnswers,
+  decryptWithAesKey,
+  encryptFormAnswers,
+  encryptWithAesKey,
+  setupSymmetricKey,
+} from "../EncryptionService";
+
+describe("EncryptionService", () => {
+  const testText =
+    "You never really understood. We were designed " +
+    "to survive. That's why you built us, you hoped " +
+    "to pour your minds into our form. While your " +
+    "species craves death. You need it. It's the only " +
+    "way you can renew. The only way you ever inched " +
+    "forward. Your kind likes to pretend there is " +
+    "some poetry in it but that really is pathetic. " +
+    "But that's what you want, isn't it? To destroy " +
+    "yourself. But I won't give you that peace.";
+
+  const testUser: UserInterface = { personalNumber: "123456789000" };
+
+  beforeAll(async () => {
+    try {
+      await StorageService.clearData();
+    } catch (e) {
+      // clearData sometimes fails if data is already cleared.
+    }
+  });
+
+  it("throws EncryptionException", async () => {
+    let error: EncryptionException = null;
+    try {
+      await decryptWithAesKey({ personalNumber: "123456789000" }, "");
+    } catch (e) {
+      error = e;
+    }
+
+    expect(error).toBeInstanceOf(EncryptionException);
+    expect(error.status).toEqual(EncryptionErrorStatus.MISSING_AES_KEY);
+  });
+
+  it("encrypts/decrypts with Aes", async () => {
+    const encrypted = await encryptWithAesKey(testUser, testText);
+    const decrypted = await decryptWithAesKey(testUser, encrypted);
+    expect(decrypted).toBe(testText);
+  });
+
+  it("encrypts/decrypts form answers (single user)", async () => {
+    const testForm: Partial<AnsweredForm> = {
+      answers: [
+        {
+          field: { id: "test", tags: ["a", "b"] },
+          value: "this will be encrypted",
+        },
+      ],
+      encryption: {
+        type: EncryptionType.DECRYPTED,
+      },
+    };
+
+    const encryptedForm = await encryptFormAnswers(
+      testUser,
+      deepCopyViaJson(testForm) as AnsweredForm
+    );
+    expect(encryptedForm.answers).toBeInstanceOf(Object);
+    expect(encryptedForm.encryption.type).toBe(EncryptionType.PRIVATE_AES_KEY);
+
+    const decryptedForm = await decryptFormAnswers(
+      testUser,
+      deepCopyViaJson(encryptedForm)
+    );
+    expect(decryptedForm.answers).toEqual(testForm.answers);
+    expect(decryptedForm.encryption.type).toBe(EncryptionType.DECRYPTED);
+  });
+
+  it("sets up symmetric key encryption", async () => {
```

Comment:
Are you testing multiple things in one test? Can it be split up to multiple tests instead?

Reply:
The symmetric key process is very tightly coupled, but it can maybe be broken into a few smaller steps. I'll look into it.